### PR TITLE
feat: add session id to all session logs (+python runner)

### PR DESCRIPTION
**Pull request by efiShtain**
- Use zapio writer so python runner logs add session_id to every log
- pass session_id to runtimes

### CODE REVIEW

File: cmd/ak/cmd/runtimes/run.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func run(ctx context.Context, b *sdkbuildfile.BuildFile, path string) (map[strin
 		defer cancel()
 	}
 
-	run, err := runtimes().Run(ctx, sdktypes.NewRunID(), path, b, nil, cbs)
+	run, err := runtimes().Run(ctx, sdktypes.NewRunID(), sdktypes.InvalidSessionID, path, b, nil, cbs)
```
Comment (tebeka) [inline cmd/ak/cmd/runtimes/run.go:138]: Will this generate a new valid session ID? And will the runtime pass it to Python?

File: runtimes/pythonrt/local_runner.go [hunk 2]
```diff
@@ -126,9 +136,11 @@ func (r *LocalPython) Start(pyExe string, tarData []byte, env map[string]string,
 		"--code-dir", r.userDir,
 	)
 	cmd.Env = overrideEnv(env, r.runnerDir, r.userDir)
+
 	if r.logRunnerCode {
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
+		r.runnerLogger = &zapio.Writer{Log: r.log, Level: zap.InfoLevel}
```
Comment (tebeka) [inline runtimes/pythonrt/local_runner.go]: In the `remote` project we have a `Log` RPC endpoint. The idea was to add a log handler to the Python logger that will send RPC calls with logs. This way you'll get log levels as well. These logs are meant for the runner only and not the user code. But currently they go to stderr/stdout and will be captured by your logger.

IMO have 2 loggers, one for stdout and one for stderr, each with a `stream` field and also the runtime.

**Review by tebeka — CHANGES_REQUESTED** [review]

File: cmd/ak/cmd/runtimes/run.go (hunk 1 above)
Comment (efiShtain) [reply cmd/ak/cmd/runtimes/run.go:138]: No real sessionID for cli call for run. It's like a manual run. So we don't really care about session (there is no session)

Comment (tebeka) [reply cmd/ak/cmd/runtimes/run.go:138]: OK

File: runtimes/pythonrt/runner/log.py [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
-    datefmt="%Y-%m-%dT%H:%M:%S",
-)
+class JsonFormatter(logging.Formatter):
+    """Formatter to dump error message into JSON"""
 
-_log = logging.getLogger("runner")
+    def format(self, record: LogRecord) -> str:
+        record_dict = {
+            "level": record.levelname,
+            "date": self.formatTime(record),
```
Comment (tebeka) [inline runtimes/pythonrt/runner/log.py:12]: Not needed, we'll have it from zap

File: runtimes/pythonrt/local_runner.go [hunk 4]
```diff
@@ -126,9 +136,11 @@ func (r *LocalPython) Start(pyExe string, tarData []byte, env map[string]string,
...
+		cmd.Stdout = r.runnerLogger
```
Comment (tebeka) [inline runtimes/pythonrt/local_runner.go]: Different logger for each with a field.

File: runtimes/pythonrt/runner/log.py [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
+            "message": record.getMessage(),
+            "module": record.module,
```
Comment (tebeka) [inline runtimes/pythonrt/runner/log.py]: I prefer the file name, not the module.

**Review by tebeka — APPROVED** [review]

**Pull request merged**